internal/manifests/manifestutils/labels.go

```diff
@@ -88,3 +89,15 @@ func TASelectorLabels(instance v1alpha1.TargetAllocator, component string) map[s
 	}
 	return selectorLabels
 }
+
+// AddExtraLabels adds extra labels to the ServiceMonitor labels map.
+func AddExtraLabels(logger *logr.Logger, labels *map[string]string, extraLabels map[string]string) {
```

Comment:
```suggestion
func AddExtraLabels(logger *logr.Logger, labels map[string]string, extraLabels map[string]string) {
```

Maps are already reference types, you don't need a pointer here.